### PR TITLE
Refactor EVM.Gas module

### PR DESCRIPTION
**Pull request by masonforest**
* Fix warnings

### CODE REVIEW

File: lib/evm/operation.ex [hunk 1]
```diff
@@ -273,6 +273,24 @@ defmodule EVM.Operation do
     end
   end
 
+  @doc """
+  Returns an operation's inputs
```
Comment (hayesgm) [inline lib/evm/operation.ex]: cool

File: lib/math_helper.ex [hunk 2]
```diff
@@ -59,4 +59,41 @@ defmodule MathHelper do
...
+      iex> MathHelper.integer_byte_size(1)
+      1
+
+      iex> MathHelper.integer_byte_size(0xfffffffff)
+      5
+
+  """
+  @spec integer_byte_size(number()) :: number()
+  def integer_byte_size(n) when n == 0, do: 0
+  def integer_byte_size(n), do: byte_size(:binary.encode_unsigned(n))
```
Comment (hayesgm) [inline lib/math_helper.ex]: What about `n |> :math.log2 |> Kernel./(8) |> :math.ceil |> round`? This should be significantly faster than actually encoding very large numbers.

File: lib/math_helper.ex [hunk 3]
```diff
@@ -59,4 +59,41 @@ defmodule MathHelper do
...
+
+      iex> MathHelper.bits_to_bytes(9)
+      1
+
+      iex> MathHelper.bits_to_bytes(256)
+      8
+
+  """
+  @spec bits_to_bytes(number()) :: number()
+  def bits_to_bytes(n), do: round(:math.ceil(n / 32))
```
Comment (hayesgm) [inline lib/math_helper.ex]: Shouldn't this be `n / 8`, not `n / 32`?

Comment (hayesgm) [reply lib/math_helper.ex]: (or maybe this is `bits_to_words`?)

Comment (masonforest) [reply lib/math_helper.ex]: Ah, yep! It's `bit_to_words`: [Renamed!](https://github.com/exthereum/evm/pull/15/commits/c39f48f89e5cb0ffc6f749774a6c038498cb6a62)

File: lib/math_helper.ex (hunk 2 above)
Comment (masonforest) [reply lib/math_helper.ex]: Hmmm, I futzed with this for a while but couldn't get it to work.
I tried:

`n |> :math.log2 |> Kernel./(8) |> :math.ceil |> round`

and

`n |> :math.log2 |> Kernel./(8) |> :math.ceil |> round |> Kernel.+(1)`

And various +1s and -1s all of which failed some common tests. Possibly it's a rounding error?

I agree converting to binary just to get the byte size seems silly and if we are going to be using these a lot optimizations are probably worth it. Alternatively maybe we could do this with `Bitwise` operators?

I'm going to merge as is and we can come back to it. Feel free to take a whack if you like.

Comment (hayesgm) [reply lib/math_helper.ex]: On second thought: there's a good chance that `:binary.encode_unsigned/1` is not a very expensive operation. Integers are obviously stored in bytes in the first place, and that operation may just wrap those bytes in a different data structure. 👍

**Pull request merged**